WordPress/gutenberg
# [RNMobile] Image Block: Improve Text Entry for Long Alt Text

Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/922

`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3236

## Description

With this PR, a new `AltTextSettings` component has been introduced to replace the current `TextControl` that exists for entering an image's Alt text. This new component makes use of `BottomSheet.SubSheet` so that users are taken to a new subsheet when entering alt text.

The end goal is to provide a better experience for users who are entering a longer alt text for their images, as the current text entry has limited room/options for scrolling.

## How has this been tested?


The general steps for testing, for both iOS and Android, are as follows:

1. Open **Image Block** settings.
2. Tap **Alt Text** and confirm that a new sub sheet opens with a text editor.
3. Enter text into editor and tap the **back** arrow.
4. Confirm that the image's alt text has been correctly updated by viewing the placeholder displayed alongside **Alt Text** in the settings panel. This can be further verified by saving changes and confirming the alt text has been correctly updated on the web.

As this is a UI change, dark mode and landscape orientation are also considerations when testing.

## Screenshots 

| Light mode  | Dark mode |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/110466861-2fe3ab00-80ce-11eb-9d77-a0563cb2c244.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/110466988-5b669580-80ce-11eb-98c1-c7c8c8fe03bf.png" width="100%"> |

## Types of changes

Non-breaking change. No additional functionality is introduced beyond UI enhancement.

## Checklist:
